src/components/Header.tsx

```diff
@@ -0,0 +1,99 @@
+import React from 'react';
+import {
+  makeStyles, withStyles, createStyles, Theme,
+} from '@material-ui/core/styles';
+import AppBar from '@material-ui/core/AppBar';
+import Toolbar from '@material-ui/core/Toolbar';
+import Typography from '@material-ui/core/Typography';
+import Button from '@material-ui/core/Button';
+import Switch from '@material-ui/core/Switch';
+import Grid from '@material-ui/core/Grid';
+
+interface ToggleMode {
+  ClickHandler: (event: React.ChangeEvent<HTMLInputElement>) => void
+}
```

Comment:
The main interface should just be named `Props` and in react, 99% of the time prop names are camel case, also `toggleTheme` is a much better name here

```suggestion
interface Props {
  toggleTheme: (event: React.ChangeEvent<HTMLInputElement>) => void
}
```